deployer/src/deployer/upload.py

```diff
@@ -1,466 +1,401 @@
-        return repr(self.key)
-
-    def set_file_hash(self):
-        with open(self.file_path, "rb") as f:
-            self.file_hash = hashlib.md5(f.read()).hexdigest()
-
-
-def upload_site(directory, config):
-    if isinstance(directory, str):
-        directory = Path(directory)
-    if not config.get("name"):
-        raise NotImplementedError("You gotta pick a name yourself")
-        # try:
-        #     repo = _find_git_repo(directory)
-        # except NoGitDirectory:
-        #     raise NoGitDirectory(
-        #         f"From {directory} can't find its git root directory "
-        #         "which is needed to supply a default branchname."
-        #     )
-        # active_branch = repo.active_branch
-        # if active_branch == "master" and config["lifecycle_days"]:
-        #     warning(
-        #         f"Warning! You're setting a lifecycle_days "
-        #         f"({config['lifecycle_days']} days) on a build from a 'master' repo."
-        #     )
-        # config["name"] = DEFAULT_NAME_PATTERN.format(
-        #     username=getpass.getuser(),
-        #     branchname=active_branch.name,
-        #     date=datetime.datetime.utcnow().strftime("%Y%m%d"),
-        # )
-        # if not config.replace("-", "").strip():
-        #     raise ValueError("Empty prefix name")
-    info(
-        f"About to upload {ppath(directory)} to prefix {config['name']!r} "
-        f"into bucket {config['bucket']!r}"
-    )
+        return f"UploadFileTask({self.file_path}, {self.key})"
 
-    print("CONFIG", repr(config))
-    return 0
-
-    session = boto3.Session(profile_name=AWS_PROFILE)
-    s3 = session.client("s3")
-
-    # First make sure the bucket exists
-    try:
-        s3.head_bucket(Bucket=config["bucket"])
-        info(f"Bucket {config['bucket']!r} exists")
-    except ClientError as error:
-        # If a client error is thrown, then check that it was a 404 error.
-        # If it was a 404 error, then the bucket does not exist.
-        if error.response["Error"]["Code"] != "404":
-            print(error.response)
-            raise
-
-        # Needs to be created.
-        bucket_config = {}
-        if config["bucket_location"]:
-            bucket_config["LocationConstraint"] = config["bucket_location"]
-        if config["dry_run"]:
-            raise CantDryRunError(
-                f"The bucket ({config['bucket']} doesn't exist and won't be created "
-                "in dry-run mode. But it needs to exist to be able to find out "
-                "what files already exist."
-            )
-        s3.create_bucket(
-            Bucket=config["bucket"],
-            ACL="public-read",
-            CreateBucketConfiguration=bucket_config,
+    def __str__(self):
+        return self.key
+
+    @property
+    def size(self):
+        return self.file_path.stat().st_size
+
+    @property
+    def etag(self):
+        """
+        Calculates and returns a value equivalent to the file's AWS ETag value.
```

Comment:
I hear you, but in practice they probably will never (or at least very rarely) change it, and it seems better to take the risk and not waste the time calculating/storing our own.

Reply:
If nobody has pushed almost any changes to Yari's source code in a 24h span, 99% of the Is-it-in-S3-already-or-shall-I-bother-to-upload it will be 99% times benefitting from comparing the ETag since the filename and filesize (both from `list_objects_v2`) will be identical. AND in the cases in the cases where the only thing changed in the 24h span is a line of SCSS or .tsx or a JS package, then again this code will be a godsend since we'll get the same filename and same filesize but a different ETag value. 
So, having it, despite its "risks" is a good idea.